connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/InboundConnectorBeanDefinitionProcessor.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.inbound;
+
+import static org.springframework.beans.factory.config.BeanDefinition.*;
+
+import io.camunda.connector.api.annotation.InboundConnector;
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.runtime.core.config.InboundConnectorConfiguration;
+import io.camunda.connector.runtime.core.inbound.InboundConnectorFactory;
+import io.camunda.connector.runtime.inbound.util.AnnotationUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class InboundConnectorBeanDefinitionProcessor
+    implements BeanDefinitionRegistryPostProcessor {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(InboundConnectorBeanDefinitionProcessor.class);
+  private final List<InboundConnectorProperties> preparedProperties = new ArrayList<>();
+
+  @Override
+  public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
+      throws BeansException {
+    ListableBeanFactory listableBeanFactory = (ListableBeanFactory) registry;
+    String[] handlerBeans =
+        listableBeanFactory.getBeanNamesForType(InboundConnectorExecutable.class);
```

Comment:
That should be
```suggestion
        listableBeanFactory.getBeanNamesForType(InboundConnectorExecutable.class, true, false);
```
to avoid initializing beans at that early stage which is forbidden as it could cause problems.

> A BeanFactoryPostProcessor may interact with and modify bean definitions, **but never bean instances. Doing so may cause premature bean instantiation, violating the container and causing unintended side effects**. If bean instance interaction is required, consider implementing [BeanPostProcessor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanPostProcessor.html) instead.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/config/BeanFactoryPostProcessor.html

Reply:
Done